### PR TITLE
Run eth block

### PR DESCRIPTION
**Pull request by ClementWalter**

### CODE REVIEW

File: cairo/src/interpreter.cairo
Comment (ClementWalter) [inline cairo/src/interpreter.cairo:1]: The diff is not properly rendered but this is mainly removing the useless `with state {` indentation as the state is an implicit arg

File: cairo/programs/os.cairo [hunk 1]
```diff
@@ -69,6 +69,11 @@ func apply_transactions{
     state: model.State*,
 }(txs_len: felt, tx_encoded: model.TransactionEncoded*) {
     alloc_locals;
+    %{
+        print(f"txs_len: {ids.txs_len}")
+        print(f"current_step: {current_step}")
+    %}
```
Comment (enitrat) [inline cairo/programs/os.cairo:75]:
```suggestion
```

File: cairo/src/utils/utils.cairo [hunk 2]
```diff
@@ -788,73 +788,161 @@ namespace Helpers {
...
+                assert valid_jumpdest.prev_value = 0;
+            }
+            return ();
         } else {
-            tempvar valid_jumpdests = valid_jumpdests;
-            tempvar next_i = next_i;
-            tempvar range_check_ptr = range_check_ptr;
+            with_attr error_message("assert_valid_jumpdest: invalid jumpdest") {
+                assert valid_jumpdest.prev_value = 1;
+            }
```
Comment (enitrat) [inline cairo/src/utils/utils.cairo:869]: if valid_jumpdest.prev_value = 1 => assert bytecode_at_jumpdest == 0x5b

File: cairo/ethereum/cancun/vm/exceptions.cairo [hunk 3]
```diff
@@ -1,9 +1,9 @@
-from ethereum_types.bytes import Bytes
+from ethereum_types.bytes import BytesStruct
```
Comment (enitrat) [inline cairo/ethereum/cancun/vm/exceptions.cairo:1]: revert to using `Bytes`

File: cairo/ethereum/cancun/vm/stack.cairo [hunk 4]
```diff
@@ -25,7 +25,7 @@ func pop{stack: Stack}() -> (U256, StackUnderflowError) {
     alloc_locals;
     let len = stack.value.len;
     if (len == 0) {
-        tempvar err = StackUnderflowError(Bytes(new BytesStruct(cast(0, felt*), 0)));
+        tempvar err = StackUnderflowError(new BytesStruct(cast(0, felt*), 0));
```
Comment (enitrat) [inline cairo/ethereum/cancun/vm/stack.cairo:28]: revert to using Bytes

File: cairo/src/interpreter.cairo [hunk 5]
```diff
@@ -986,25 +960,29 @@ namespace Interpreter {
...
         let actual_fee = total_gas_used * env.gas_price;
         let (fee_high, fee_low) = split_felt(actual_fee);
         let actual_fee_u256 = Uint256(low=fee_low, high=fee_high);
         let transfer = model.Transfer(env.origin, env.coinbase, actual_fee_u256);
 
-        with state {
-            State.add_transfer(transfer);
-            State.finalize();
-        }
+        State.add_transfer(transfer);
```
Comment (enitrat) [inline cairo/src/interpreter.cairo:984]: reminder that this will need to be burnt

File: cairo/tests/fixtures/runner.py [hunk 6]
```diff
@@ -344,7 +344,6 @@ def _factory(entrypoint, *args, **kwargs):
         if add_output:
             final_output = serde.serialize_list(output_ptr)
 
-        cumulative_retdata_offsets = serde.get_offsets(return_data_types)
```
Comment (enitrat) [inline cairo/tests/fixtures/runner.py:347]: variable missing

**Review by enitrat — CHANGES_REQUESTED** [review]

File: cairo/tests/fixtures/runner.py (hunk 6 above)
Comment (ClementWalter) [reply cairo/tests/fixtures/runner.py:347]: it was a redefinition, same line few lines above

**Review by enitrat — APPROVED** [review]

**Pull request merged**